src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -158,25 +160,33 @@ public LinkedPointSet(int[] edges, int[] distances0_mm, int[] distances1_mm, Poi
 
     /**
      * Associate the points in this PointSet with the street vertices at the ends of the closest street edge.
-     * @param relinkZone only link points inside this geometry in FIXED POINT DEGREES, leaving all the others alone. If null, link all points.
+     * @param all If true, link all points, otherwise link only those that were previously connected to edges that have
+     *            been deleted (i.e. split).
+     *            We will need to change this behavior when we allow creating new edges rather than simply splitting
+     *            existing ones.
      */
-    private void linkPointsToStreets(Geometry relinkZone) {
+    private void linkPointsToStreets(boolean all) {
         LambdaCounter counter = new LambdaCounter(LOG, pointSet.featureCount(), 10000,
                 "Linked {} of {} PointSet points to streets.");
         // Perform linkage calculations in parallel, writing results to the shared parallel arrays.
         IntStream.range(0, pointSet.featureCount()).parallel().forEach(p -> {
-            // When working with a scenario, skip all points outside the zone that could be affected by new edges.
-            // The linkage from the original base StreetNetwork will be retained for these points.
-            if (relinkZone != null && !relinkZone.contains(pointSet.getJTSPointFixed(p))) return;
-            Split split = streetLayer.findSplit(pointSet.getLat(p), pointSet.getLon(p), MAX_OFFSTREET_WALK_METERS, streetMode);
-            if (split == null) {
-                edges[p] = -1;
-            } else {
-                edges[p] = split.edge;
-                distances0_mm[p] = split.distance0_mm;
-                distances1_mm[p] = split.distance1_mm;
+            // When working with a scenario, skip all points that are not linked to a deleted street (i.e. one that has
+            // been split). At the current time, the only street network modification we support is splitting existing streets,
+            // so the only way a point can need to be relinked is if it is connected to a street which was split (and therefore deleted).
+            // FIXME when we permit street network modifications beyond adding transit stops we will need to change how this works,
+            // we may be able to use some type of flood-fill algorithm in geographic space, expanding the relink envelope until we
+            // hit edges on all sides or reach some predefined maximum.
+            if (all || streetLayer.edgeStore.temporarilyDeletedEdges != null && streetLayer.edgeStore.temporarilyDeletedEdges.contains(edges[p])) {
```

Comment:
I would lean toward grouping the sub-expression after || with parentheses rather than relying on the reader's operator precedence knowledge.